src/Common/ShellCommand.cpp

```diff
@@ -243,11 +248,66 @@ std::unique_ptr<ShellCommand> ShellCommand::executeImpl(
         sigprocmask(0, nullptr, &mask);
         sigprocmask(SIG_UNBLOCK, &mask, nullptr);
 
+        close(mappings_ready.fds_rw[1]);
+        char c;
+        read(mappings_ready.fds_rw[0], &c, sizeof(c));
+
+        if (setuid(1000) < 0) {
+            _exit(static_cast<int>(ReturnCodes::CANNOT_SETUID));
+        }
+        if (setgid(1000) < 0) {
+            _exit(static_cast<int>(ReturnCodes::CANNOT_SETGID));
+        }
+        if (setgroups(0, nullptr) < 0) {
+            _exit(static_cast<int>(ReturnCodes::CANNOT_SETGROUPS));
+        }
         execv(filename, argv);
         /// If the process is running, then `execv` does not return here.
 
         _exit(static_cast<int>(ReturnCodes::CANNOT_EXEC));
+    });
+
+    PipeFDs pipe_sync;
+
+    const size_t stack_sz = 16 * 1024;
+    auto stack = std::make_unique<unsigned char[]>(stack_sz);
+    // !! add config option to run isolated and non-isolated processes
+    // pid_t pid = reinterpret_cast<pid_t(*)()>(real_vfork)();
+    pid_t pid = clone(execInFork, stack.get() + stack_sz, CLONE_VM | CLONE_NEWUSER | SIGCHLD, &child_func);
```

Comment:
Executable UDF tests in CI fail with an error:
```
Code: 299. DB::Exception: Received from localhost:9000. DB::ErrnoException. DB::ErrnoException: Cannot clone, errno: 1, strerror: Operation not permitted: While processing test_function(toUInt64(2), toUInt64(2)). (CANNOT_FORK)
```
Seems like clone with certain options [doesn't work in unprivileged comainers](https://stackoverflow.com/a/32126393), maybe this is the problem